galley/pkg/config/analysis/analyzers/analyzers_test.go

```diff
@@ -45,10 +45,11 @@ type message struct {
 }
 
 type testCase struct {
-	name       string
-	inputFiles []string
-	analyzer   analysis.Analyzer
-	expected   []message
+	name           string
+	inputFiles     []string
+	meshConfigFile string // Optional
```

Comment:
Would you consider inlining the meshconfig here as a string constant rather than a filename? I imagine in general, we'll be specifying single mesh config values rather than a complicated structure, and being able to see the actual mesh config values alongside the test case would be pretty valuable.

Reply:
It's something I initially considered, but moved away from. 

My bias was towards "easiest implementation" that doesn't require creating temp files or changing the SourceAnalyzer API, especially since we don't have a lot of specific use-cases here yet to guide our approach. 

I also like that how we specify mesh config is consistent with how we specify the input YAML. With an inlined value, if we needed more than one test case using the same mesh config, we'd either need to 1) repeat it, or 2) extract a named const, which has about the same level of indirection as referring to a named file.

Reply:
I would be totally fine repeating it rather than pulling it into a constant (which, as you mentioned, impairs readability). It's pretty common in tests to repeat values as tests must often be understood in isolation rather than the entire package as a whole.

You gave a better reason not to do this, though, which is that the SourceAnalyzer apparently insists on files as input (and doesn't support streams). If that's true then your approach makes total sense. Thanks for explaining :)